### PR TITLE
fix: compiler warning

### PR DESCRIPTION
**Pull request by Groveer**

### CODE REVIEW

**Review by BLumia — APPROVED** [review]

File: src/server/kernel/wseat.h [hunk 1]
```diff
@@ -53,6 +53,7 @@ class WSeat : public WServerInterface, public WObject
     W_DECLARE_PRIVATE(WSeat)
     Q_PROPERTY(WInputDevice* keyboard READ keyboard WRITE setKeyboard NOTIFY keyboardChanged FINAL)
 public:
+    using QObject::eventFilter;
```
Comment (zccrs) [inline src/server/kernel/wseat.h:56]: Why?

File: src/server/qtquick/private/wquickxdgdecorationmanager.cpp [hunk 2]
```diff
@@ -34,13 +34,13 @@ class WQuickXdgDecorationManagerPrivate : public WObjectPrivate
     void updateDecorationMode(QWXdgToplevelDecorationV1 *decorat);
 
     WQuickXdgDecorationManager::DecorationMode modeBySurface(WSurface *surface) const {
-        return decorations.value(surface, WQuickXdgDecorationManager::DecidesByClient);
```
Comment (zccrs) [inline src/server/qtquick/private/wquickxdgdecorationmanager.cpp:37]: @wineee

File: src/server/qtquick/private/wquickxdgdecorationmanager.cpp [hunk 3]
```diff
@@ -34,13 +34,13 @@ class WQuickXdgDecorationManagerPrivate : public WObjectPrivate
...
+        return decorations.value(surface, WQuickXdgDecorationManager::Undefined);
     }
 
     W_DECLARE_PUBLIC(WQuickXdgDecorationManager)
 
     QWXdgDecorationManagerV1 *manager = nullptr;
-    WQuickXdgDecorationManager::DecorationMode mode = WQuickXdgDecorationManager::DecidesByClient;
```
Comment (zccrs) [inline src/server/qtquick/private/wquickxdgdecorationmanager.cpp:43]: @wineee

File: src/server/qtquick/private/wquickxdgdecorationmanager.cpp [hunk 4]
```diff
@@ -63,8 +63,17 @@ void WQuickXdgDecorationManagerPrivate::updateDecorationMode(QWXdgToplevelDecora
...
+    WQuickXdgDecorationManager::DecorationMode mode = WQuickXdgDecorationManager::Undefined;
+    switch (decorat->handle()->requested_mode) {
+        case WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_NONE:
+            mode = WQuickXdgDecorationManager::None;
+            break;
+        case WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_CLIENT_SIDE:
+            mode = WQuickXdgDecorationManager::Client;
+            break;
+        case WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE:
+            mode = WQuickXdgDecorationManager::Server;
```
Comment (zccrs) [inline src/server/qtquick/private/wquickxdgdecorationmanager.cpp:75]: break;

File: src/server/qtquick/private/wquickxdgdecorationmanager_p.h [hunk 5]
```diff
@@ -44,9 +44,10 @@ class WAYLIB_SERVER_EXPORT WQuickXdgDecorationManager : public WQuickWaylandServ
     explicit WQuickXdgDecorationManager(QObject *parent = nullptr);
 
     enum DecorationMode {
```
Comment (zccrs) [inline src/server/qtquick/private/wquickxdgdecorationmanager_p.h:46]: @wineee

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**